packages/hl7/src/connection.ts

```diff
@@ -81,15 +98,6 @@ export class Hl7Connection extends Hl7Base {
       if (this.enhancedMode) {
         this.send(event.message.buildAck({ ackCode: 'CA' }));
       }
-      // Check if we have messages pending a response in the pending messages map
-      if (!this.pendingMessages.size) {
```

Comment:
Removed this because this actually causes issues when receiving messages on a server-side connection since server-side connections don't have pending messages